ray_lightning/launchers/ray_launcher.py

```diff
@@ -0,0 +1,360 @@
+from typing import Callable, List, Any, Tuple, Optional, \
+    NamedTuple, Dict
+
+from collections import defaultdict
+from contextlib import closing
+import os
+import socket
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies.launchers import _Launcher
+from pytorch_lightning.trainer.states import TrainerState
+from pytorch_lightning.utilities.apply_func import apply_to_collection,\
+    move_data_to_device
+from pytorch_lightning.utilities.types import _PATH
+import numpy as np
+import torch
+
+import ray
+from ray import ObjectRef
+from pytorch_lightning.utilities.rank_zero import rank_zero_debug
+from ray.util.queue import Queue
+
+from ray_lightning.session import init_session
+from ray_lightning.util import process_results, to_state_stream, \
+    load_state_stream
+from ray_lightning.tune import TUNE_INSTALLED, is_session_enabled
+from pytorch_lightning.strategies import Strategy
+
```

Comment:
For both the launchers, can we distinguish between what code gets run on the driver vs. what code gets run on the workers?

Reply:
good idea

Reply:
done